src/eos-updater-fetch.c

```diff
@@ -113,27 +139,30 @@ content_fetch (GTask *task,
       goto error;
     }
 
-  message ("Fetch: %s:%s resolved to: %s", src, ref, sum);
+  message ("Fetch: %s:%s resolved to: %s", remote, ref, sum);
+  progress = ostree_async_progress_new_and_connect (update_progress, updater);
+
+  if (data->overridden_urls)
+    {
+      guint idx = g_random_int_range (0, g_strv_length (data->overridden_urls));
 
+      url_override = data->overridden_urls[idx];
```

Comment:
Yeah, kinda. When polling for updates on the LAN, we can get several servers that have the same, latest update. We just choose one randomly.